### ISSUE TITLE
simpler explode and join

### ISSUE THREAD

**Issue by adamlerer** [pytorch/pytorch#427]
Is there or can we add something like chunk that selects individual rows? It's especially common for autograd because indexing is inefficient (and I'm assuming iterating is inefficient too? But I'm not sure... but even if it's not, it can only do dim 0).

This is the boilerplate:
```
for row in torch.chunk(x, x.size(0)):
   row = row.squeeze(0)
```

The join operation is also annoying:

```
x = torch.cat([e.unsqueeze(0) for e in x])
```

**Comment by apaszke** [comment]
Added `torch.unbind` in #661.
